src/code.rs

```diff
@@ -441,6 +442,51 @@ impl {struct_name} {{
     }}
 "##));
 
+    // Table::filter() helper fn
+    {
+        let diesel_backend = &config.diesel_backend;
+        let filters = table
+            .columns
+            .iter()
+            .map(|column| {
+                let column_name = column.name.to_string();
+                format!(
+                    r##"
+        if let Some(filter_{column_name}) = filter.{column_name}.clone() {{
+            query = query.filter({schema_path}{table_name}::{column_name}.eq(filter_{column_name}));
+        }}"##
+                )
+            })
+            .collect::<Vec<_>>()
+            .join("");
+        buffer.push_str(&format!(
+            r##"
+    /// A utility function to help build custom search queries
+    /// 
+    /// Example:
+    /// 
+    ///     ```rust
+    ///     // create a filter for completed todos
+    ///     let query = Todo::filter(TodoFilter {{
+    ///       completed: Some(true),
+    ///       ..Default::default()
+    ///     }});
```

Comment:
```suggestion
    ///     let query = Todo::filter(TodoFilter {
    ///       completed: Some(true),
    ///       ..Default::default()
    ///     });
```

did i miss something or why are there `{{` and `}}`?

Reply:
I think it's because my editor was being weird about it haha
<img width="499" alt="image" src="https://github.com/Wulf/dsync/assets/4259838/aca078f4-2bb1-4999-8430-3d99629b235c">

Reply:
ah nevermind, it's because it was in a `format!()`. Here I was thinking why this simple doc comment wasn't working!

In format!(), you need to escape '{' usage by using two of them. Hence the {{ }}. We do this a lot in dsync!

Reply:
right, also forgot it was inside a `format!()`, i was just skimming through with the github editor (no in-detail highlighting or playing around with some changes)